lib/widgets/components/chip_component.dart

```diff
@@ -0,0 +1,123 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_internship/helpers/spacer_width.dart';
+
+enum ChipType { Flat, Clean }
+
+class ChipComponent extends StatelessWidget {
+  final ChipType chipType;
+  final IconData? icon;
+  final String label;
+
+  const ChipComponent({
+    Key? key,
+    required this.chipType,
+    this.icon,
+    required this.label,
+  }) : super(key: key);
+
+  @override
+  Widget build(BuildContext context) {
+    switch (chipType) {
+      case ChipType.Flat:
+        return FlatChip(label: label, icon: icon);
+      case ChipType.Clean:
+        return CleanChip(label: label, icon: icon);
+    }
+  }
+}
+
+//sivi chip
+class FlatChip extends StatelessWidget {
+  final String label;
+  final IconData? icon;
+
+  const FlatChip({
+    super.key,
+    required this.label,
+    this.icon,
+  });
+
+  @override
+  Widget build(BuildContext context) {
+    return Chip(
+      label: Align(
+        alignment: Alignment.center,
+        child: Padding(
+          padding: const EdgeInsets.symmetric(vertical: 8, horizontal: 12),
+          child: Row(
```

Comment:
This part is same in both Flat and Clean component widget vise, make it more reusable, make it as separate private widget.

Reply:
Why is there Align.

Reply:
👎